docs/docsite/rst/dev_guide/testing/sanity/no-underscore-variable.rst

```diff
@@ -0,0 +1,28 @@
+Sanity Tests » no-underscore-variable
+=====================================
+
+In the future, Ansible may use the identifier ``_`` to internationalize its
+message strings.  To be ready for that, we need to make sure that there are
+no conflicting identifiers defined in the code base.
+
+In common practice, ``_`` is frequently used as a dummy variable (a variable
+to receive a value from a function where the value is useless and never used).
+In Ansible, we're using the identifier ``dummy`` for this purpose instead.
```

Comment:
May want to mention cases where static checking tool complains about multiple uses of 'dummy'. Pylint will complain if 'dummy' is used in a scope, then a list comprehension in that scope uses 'dummy' as well'.

ie, something like:

``` python
def foobar(the_stuff, misc=None):
   stuff, dummy = parse_stuff(the_stuff)

   filtered_stuff = [misc_thing for misc_thing, dummy in parse_misc_to_tuple(misc)]
```
If we want to detail that case, and suggest an option (I just used '_dummy', but whatever we decide...).